src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -1103,21 +1132,55 @@ inline void BroadcastComputeImpl(const nnvm::NodeAttrs& attrs,
           out_shape[i] = 1;
         }
       }
-      if (dst_shape.ndim() == 2) {
+      uint64_t axes[dst_shape.ndim()], out_stride[dst_shape.ndim()];
+      int iter = dst_shape.ndim() - 1, i = 0;
+      bool flag = false;
```

Comment:
what does the flag here mean? Please use a more informative name.

Reply:
this flag ensure if broadcast_axis is supplied with input where input shape = output shape(i.e. no broadcast). In such a case we are simply invoking copy